features/user/service/service.go

```diff
@@ -95,7 +95,12 @@ func Delete(id string) error {
 		return errors.New("you must login first")
 	}
 
-	err := repository.DeleteAccount(id)
+	result, err := GetProfile(id)
+	if err != nil {
+		return err
+	}
+
+	err = repository.DeleteAccount(result.ID)
```

Comment:
why is this changed?

Reply:
because when I tried to fill the id in the endpoint with a random string (id that is not in the database), the status became 204. not an error. So I fixed it